distributed/semaphore.py

```diff
@@ -460,6 +462,28 @@ def acquire(self, timeout=None):
         timeout = parse_timedelta(timeout)
         return self.client.sync(self._acquire, timeout=timeout)
 
+    async def _release(self):
+        # popleft to release the oldest lease first
+        lease_id = self._leases.popleft()
+        logger.info("%s releases %s for %s", self.client.id, lease_id, self.name)
+
+        try:
+            await retry_operation(
+                self.client.scheduler.semaphore_release,
+                name=self.name,
+                lease_id=lease_id,
+                operation="semaphore release: client=%s, lease_id=%s, name=%s"
+                % (self.client.id, lease_id, self.name),
+            )
+            return True
+        except Exception:  # Release fails for whatever reason
+            logger.error(
+                "Release failed for client=%s, lease_id=%s, name=%s. Cluster network might be unstable?"
+                % (self.client.id, lease_id, self.name),
+                exc_info=True,
+            )
+            return False
+
     def release(self):
         """
         Release a semaphore.
```

Comment:
Please add a comment about the meaning of the return value. In particular, this should instruct users how to proceed in case of a failed release. I'm still wondering if we should return anything for the release function. After all, what the user is not supposed to do is release again in this scenario since the most likely scenario is that the connection broke up but the lease ID was already removed from internal tracking. Releasing again would cause the "Released too many times" exception